### DISCUSSION TITLE
RLS form submission for anyone

### DISCUSSION THREAD

**Discussion by jwright04** [supabase/supabase#7611] [Questions] [answered]
I have a table that I've enabled RLS on but I'm unable to get data to the form due to an RLS error.  This is a public form which requires no authentication for insert. I used a template with the following

```
Policy name: "Enable insert for any user"
Allowed operation: "Insert"
Target roles: "" <--- I left blank, but also tried with "anon"
With CHECK expression: "true"
```
Is there something else I should be adding?

I referenced the following
https://supabase.com/docs/guides/auth/row-level-security#policies-with-security-definer-functions

**Comment by rustamd** [comment, marked as answer]
Insert normally requires SELECT permissions as well to return values from Postgres. You have to specify `returning: 'minimal'` to not need SELECT permissions

Edit: [Reference](https://supabase.com/docs/reference/javascript/insert#notes)

**Reply by jwright04** [reply]
Spot on, thank you!
